include/deal.II/base/types.h

```diff
@@ -41,6 +41,12 @@ namespace types
    */
   typedef unsigned int subdomain_id;
 
+  /**
+   * The type used for global indices of vertices.
+   */
+  typedef unsigned long long int global_vertex_index;
+#  define DEAL_II_VERTEX_INDEX_MPI_TYPE MPI_UNSIGNED_LONG_LONG
```

Comment:
It's up to you if we want to switch between 32 and 64-bit indices. I chose to use 64-bit because the library I plan to use always requires an unsigned long long int. Since this function is not used through the whole library, I think the gain in memory will be minimal.